ReactWindows/ReactNative/Views/WebView/ReactWebViewManager.cs

```diff
@@ -0,0 +1,289 @@
+        /// Sets the user-agent for this webview.
+        /// </summary>
+        /// <param name="view">a webview instance.</param>
+        /// <param name="userAgent"></param>
+        [ReactProp("userAgent")]
+        public void SetUserAgent(Windows.UI.Xaml.Controls.WebView view, string userAgent)
+        {
+            // No implementation
+        }
+
+        /// <summary>
+        /// Sets the JS to be injected when the webpage loads.
+        /// </summary>
+        /// <param name="view">a webview instance.</param>
+        /// <param name="injectedJavaScript"></param>
+        [ReactProp("injectedJavaScript")]
+        public void SetInjectedJavaScript(Windows.UI.Xaml.Controls.WebView view, string injectedJavaScript)
+        {
+            if (_injectedJS.ContainsKey(view.GetTag()))
+            {
+                _injectedJS.Remove(view.GetTag());
+            }
+            _injectedJS.Add(view.GetTag(), injectedJavaScript);
+        }
+
+        /// <summary>
+        /// Sets webview source.
+        /// </summary>
+        /// <param name="view">a webview instance.</param>
+        /// <param name="source"></param>
+        [ReactProp("source")]
+        public void SetSource(Windows.UI.Xaml.Controls.WebView view, JObject source)
+        {
+            if (source != null)
+            {
+                JToken html, uri;
+
+                if (source.TryGetValue("html", out html) && html.Type == JTokenType.String)
+                {
+                    JToken baseUrl;
+                    if (source.TryGetValue("baseUrl", out baseUrl) && baseUrl.Type == JTokenType.String)
+                    {
+                        view.Source = new Uri(baseUrl.Value<string>());
+                    }
+
+                    view.NavigateToString(html.Value<string>());
+                    return;
+                }
+
+                if (source.TryGetValue("uri", out uri) && uri.Type == JTokenType.String)
+                {
+                    JToken method, headers, body;
+
+                    var request = new HttpRequestMessage();
+                    request.RequestUri = new Uri(uri.Value<string>());
+
+                    if (source.TryGetValue("method", out method) && method.Type == JTokenType.String)
+                    {
+                        if (method.Value<string>().Equals("GET")) request.Method = HttpMethod.Get;
+                        else if (method.Value<string>().Equals("POST")) request.Method = HttpMethod.Post;
+                        else return;
+                    }
+                    else request.Method = HttpMethod.Get;
+
+                    if (source.TryGetValue("headers", out headers) && headers.Type == JTokenType.Object)
+                    {
+                        IEnumerator<KeyValuePair<string, JToken>> enumerator = ((JObject)headers).GetEnumerator();
+                        enumerator.Reset();
+                        while (enumerator.MoveNext())
+                        {
```

Comment:
You are right - the IEnumerator<KeyValuePair<string, JToken>> enumerator is not needed.